### PR TITLE
Added ability to filter lists by owners.

### LINKED ISSUE #7: Ability to show a user's owned tags
This issue will add a new optional argument to `/tags list` named `user`. When set, it will only show tags that are owned by that user.

Part of #1.

### PR DESCRIPTION
**Pull request by bsrecker**
Added an optional command option for a `user`.

This will filter the list to only tags owned by that user.

### CODE REVIEW

File: leaf/extensions/tags.py [hunk 1]
```diff
@@ -65,26 +65,32 @@ async def tag_autocomplete(
...
+        if user is not None:
+            query = "SELECT * FROM tags WHERE guild_id = $1 AND owner_id = $2 AND deleted = FALSE"
+            tags = await self.bot.database.fetch(query, interaction.guild.id, user.id)
+        else:
+            query = "SELECT * FROM tags WHERE guild_id = $1 AND deleted = FALSE"
+            tags = await self.bot.database.fetch(query, interaction.guild.id)
 
-        if not tags:
+        embeds = []
+        if len(tags) == 0:
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: Why not `if not tags` or `if not len(tags)`?

**Review by mudkipdev — APPROVED** [review]

File: leaf/extensions/tags.py (hunk 1 above)
Comment (bsrecker) [reply leaf/extensions/tags.py]: Ill update that.

**Comment by mudkipdev** [comment]
Also remove the pagination menu if the user does not have any tags.

File: leaf/extensions/tags.py [hunk 2]
```diff
@@ -65,26 +65,32 @@ async def tag_autocomplete(
         ]
 
     @app_commands.describe(
+        user="The user you wish to search for.",
```
Comment (mudkipdev) [inline leaf/extensions/tags.py]: I think this makes it sound like it's a required argument

**Pull request merged**